include/state_machine.h

```diff
@@ -57,6 +64,9 @@ class StateMachine
   std::list<State> states_;
   std::list<State> escape_states_;
   const State* current_state_ = nullptr;
+  const State* previous_state_ = nullptr;
+
+  bool inEscapeMenu_ = false;
```

Comment:
Please use consistent naming :wink: 
```suggestion
  bool in_escape_menu_ = false;
```

Reply:
Whoops, you're right